compliance/node/index/indexer.go

```diff
@@ -109,12 +109,14 @@ type NodeIndexerConfig struct {
 }
 
 // DefaultNodeIndexerConfig is the default configuration for a node indexer.
-var DefaultNodeIndexerConfig = NodeIndexerConfig{
-	HostPath: env.NodeIndexHostPath.Setting(),
-	// The default, mTLS-capable client will be used.
-	Client:             nil,
-	Repo2CPEMappingURL: env.NodeIndexMappingURL.Setting(),
-	Timeout:            10 * time.Second,
+func DefaultNodeIndexerConfig() NodeIndexerConfig {
+	return NodeIndexerConfig{
+		HostPath: env.NodeIndexHostPath.Setting(),
+		// The default, mTLS-capable client will be used.
+		Client:             nil,
+		Repo2CPEMappingURL: "https://" + env.AdvertisedEndpoint.Setting() + env.NodeIndexMappingURLPath.Setting(),
```

Comment:
TODO: Add task for unifying `AdvertisedEndpoint` with its copy variable. Consider further ideas:
- guessing the namespace
- remove the hardcoding of `/scanner/definitions?file=repo2cpe` - it is hardcoded in the server part in Sensor.

Reply:
The followup has been created: https://issues.redhat.com/browse/ROX-28780